db/users.go

```diff
@@ -0,0 +1,85 @@
+package db
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/lbryio/lbrytv/config"
+	"github.com/lbryio/lbrytv/lbrynet"
+	"github.com/lbryio/lbrytv/models"
+	"github.com/lbryio/lbrytv/monitor"
+
+	"github.com/lbryio/lbry.go/extras/lbryinc"
+	"github.com/volatiletech/sqlboiler/boil"
+)
+
+// TokenHeader is the name of HTTP header which is supplied by client and should contain internal-api auth_token
+const TokenHeader string = "X-Lbry-Auth-Token"
+
+// GetUserByToken retrieves user by internal-api auth_token
+// TODO: Refactor out into logical pieces
+// TODO: Implement different error types for better error messages
+func GetUserByToken(token string) (*models.User, error) {
+	var u *models.User
+	ctx := context.Background()
+
+	u, err := models.Users(models.UserWhere.AuthToken.EQ(token)).OneG(ctx)
+	if err != nil {
+		Conn.Logger.LogF(monitor.F{"token": token}).Info("token supplied but isn't present in the database, trying internal-apis")
+		c := lbryinc.NewClient(token)
+		c.ServerAddress = config.Settings.GetString("InternalAPIHost")
+		r, err := c.UserMe()
+		if err != nil {
+			Conn.Logger.LogF(monitor.F{"token": token}).Error("internal-api responded with an error")
+			// No user found in internal-apis database, give up at this point
+			return u, err
+		}
+		if email, ok := r["primary_email"].(string); ok {
+			Conn.Logger.LogF(monitor.F{"token": token, "email": email}).Info("got an account from internal-apis")
+			u, err := models.Users(models.UserWhere.AuthToken.EQ(email)).One(ctx, Conn.DB)
+			if err != nil {
+				Conn.Logger.LogF(monitor.F{"token": token, "email": email}).Info("user seen first time, creating locally")
+				// No user with that email in our database, create one with the daemon instance and save to the database
+				a, err := lbrynet.CreateAccount(email)
+				if err != nil {
+					return nil, err
+				}
+
+				u = new(models.User)
+				u.Email = email
+				u.AuthToken = token
+				u.IsIdentityVerified = false
+				u.HasVerifiedEmail = false
+				u.SDKAccountID = a.ID
+				u.PrivateKey = a.PrivateKey
+				u.PublicKey = a.PublicKey
+				u.Seed = a.Seed
+				err = u.InsertG(ctx, boil.Infer())
+
+				if err != nil {
+					Conn.Logger.LogF(monitor.F{"token": token, "email": email}).Error("error inserting a record, rolling back account")
+					_, errDelete := lbrynet.RemoveAccount(email)
+					if errDelete != nil {
+						Conn.Logger.LogF(monitor.F{"token": token, "email": email}).Errorf("error rolling back account: %v", errDelete)
+					}
+					return nil, err
+				}
+				return u, nil
+			}
+		}
+	}
+	return u, nil
+}
+
+// GetAccountIDFromRequest retrieves SDK  account_id of a user making a http request
+// by a header provided by the http client
+func GetAccountIDFromRequest(req *http.Request) (string, error) {
```

Comment:
So in our [api](https://github.com/lbryio/lbry.go/tree/master/extras/api) package do a search for `FormValues`. It's actually a really clean way for handling parameters with validators too so you dont have to do the below. We use it extensively in internal-apis. I like it alot. I use it in chainquery as well.

Reply:
`FormValues` looks good. Seems like it's just for, well, forms. I'm retrieving a header value here.

Reply:
Yeah agreed. I guess I also wasn't sure why you chose headers vs parameters. I assumed it was a trivial decision and was suggesting to use params solely to leverage the FormValues since it makes things so much easier.

Reply:
Request body contains SDK query already, there’s no space for params. And I’d rather not transfer sensitive data in the URL.